unittest/main.cpp

```diff
@@ -23,6 +23,8 @@
 
 #define DOCTEST_THREAD_LOCAL // Avoid MinGW thread_local bug
 #define DOCTEST_CONFIG_IMPLEMENT
+#include <ostream>
```

Comment:
Since the defines above configure the behavior of `third_party/doctest.h` below, I suggest moving the `ostream` inclusion above the defines.